File: day3/puzzle-2.php
<?php

function checkColl($x, $line) {
    $spot = substr($line, $x-1, 1);
    if ($spot == "#") {
        return 1;
    }
    else {
        return 0; 
    }
}


$validCount = 0;
$collisionResult = [];
$slopeList = [[1,1], [3,1], [5,1], [7,1], [1,2]];
$filename = 'input.txt';
$fp = @fopen($filename, 'r');
if ($fp) {
   $array = explode("\n", fread($fp, filesize($filename)));
   $lineCount = count($array);
}

if (!is_null($array)) {
    foreach ($slopeList as $xY) {
        $curX = 1;
        $curY = 0;
        echo "X: " . $xY[0] . " Y: " . $xY[1] . "\n";
        $slopeX = $xY[0];
        $slopeY = $xY[1];
        $collisions = 0;
        while ($curY <= $lineCount) {
            if ($curY > 0) {
                echo $curY."\n";
                $collisions += checkColl($curX, $array[$curY]);
            }
            $curY += $slopeY;
            $curX += $slopeX;
        }
        array_push($collisionResult, $collisions);
    }
}
$result = $collisionResult[0] * $collisionResult[1] * $collisionResult[2] * $collisionResult[3] * $collisionResult[4];
echo "Your multiplication result is: " . $result . "\n";

File: python/day3/puzzle-2.py

def checkCollision(x, line):
  point = line[x-1]
  if point == "#":
    return 1
  else:
    return 0

import numpy as np
inputFile = open("input.txt")
input = np.array([])
collisionCollection = np.array([])
for line in inputFile:
  input = np.append(input,line)
inputFile.close()
slopes = np.array([[1,1], [3,1], [5,1], [7,1], [1,2]])

for slopeXY in slopes:
    idxRoot = 0
    slopeX = slopeXY[0]
    slopeY = slopeXY[1]
    curX = 1
    curY = 0
    collisions = 0
    while idxRoot < len(input):
      line = input[idxRoot]
      if (curY > 0 and curY < len(input)):
        collisions += checkCollision(curX, input[curY])
      curY += slopeY
      curX += slopeX
      idxRoot += 1
    collisionCollection = np.append(collisionCollection, collisions)
answer = 1
for x in collisionCollection:
    answer = x * answer
txt = "The Answer is {}"
print(txt.format(answer))